blocks/cards/cards.js

```diff
@@ -3,17 +3,26 @@ import {
   ul, li, a, div, img, h4, i, br,
 } from '../../scripts/dom-helpers.js';
 
+let cardImage = '';
+
 export default function decorate(block) {
   /* change to ul, li */
   const $ul = ul();
   [...block.children].forEach((row) => {
+    [...row.children].forEach((col, index) => {
+      if (index === 0) {
+        cardImage = createOptimizedPicture(col.querySelector('a'));
```

Comment:
```suggestion
        cardImage = createOptimizedPicture(col.querySelector('a').href);
```
The current approach seems to work, yet I'm not sure we should pass <a> to new URL(). 
URL documentation isn't alear about passing anchor element to it https://developer.mozilla.org/en-US/docs/Web/API/URL/URL.

Reply:
Thank You @akasjain2  - Actually the intent is to convert the image link into Picture - image form right in the beginning & then keep intact the  entire below code for cards.js. For Eg. 

I have to just replace divEl.querySelector('picture')) with cardImage which has got picture - - > image details within.